hazelcast/src/main/java/com/hazelcast/client/impl/protocol/AuthenticationStatus.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2008-2015, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.client.impl.protocol;
+
+/**
+ * Status codes of authentication results
+ */
+public enum AuthenticationStatus {
+
+    AUTHENTICATED(0),
+    CREDENTIALS_FAILED(1),
+    SERIALIZATION_VERSION_MISMATCH(2);
+
+    private final byte status;
+
+    AuthenticationStatus(int status) {
+        this.status = (byte) status;
+    }
+
+    public byte getStatus() {
```

Comment:
I would call this 'id' This is done at other parts in the system else you get status.getStatus() which looks funny.